commons/protocol/src/main/scala/sbt/protocol/Values.scala

```diff
@@ -66,14 +66,14 @@ final case class TaskSuccess(value: BuildValue) extends TaskResult {
   override def isSuccess = true
   override def resultWithCustomThrowable[A, B <: Throwable](implicit unpickleResult: Unpickler[A], unpickleFailure: Unpickler[B]): Try[A] = {
     value.value[A] match {
-      case Some(v) => Success(v)
-      case None => Failure(new Exception(s"Failed to deserialize ${value.serialized}"))
+      case Success(v) => Success(v)
+      case Failure(t) => Failure(new Exception(s"Failed to deserialize ${value.serialized}", t))
     }
   }
   override def resultWithCustomThrowables[A](throwableDeserializers: ThrowableDeserializers)(implicit unpickleResult: Unpickler[A]): Try[A] = {
     value.value[A] match {
-      case Some(v) => Success(v)
-      case None => Failure(new Exception(s"Failed to deserialize ${value.serialized}"))
+      case Success(v) => Success(v)
+      case Failure(t) => Failure(new Exception(s"Failed to deserialize ${value.serialized}", t))
     }
```

Comment:
oh no, I remember now this is the success case so we ignore it and we use it in the fail case, nevermind.